### PR TITLE
Matched migration enum defs with graphql schema defs

### PR DESCRIPTION
**Pull request by OddTomBrooks**
# EASI-000

## Changes and Description

- Enums were named one way in the graphql schema but another in the migration, this PR resolves that

## How to test this change

Try updating models with all ClaimsBasedPayTypes and AnticipatedPaymentFrequencyType::SEMIMONTHLY

## PR Author Review Checklist

- [ ] Met the ticket's acceptance criteria, or will meet them in a subsequent PR.
- [ ] Added or updated tests for backend resolvers or other functions as needed.
- [ ] Added or updated client tests for new components, parent components, functions, or e2e tests as necessary.
- [ ] Tested user-facing changes with voice-over and the [rotor menu](https://support.apple.com/guide/voiceover/with-the-voiceover-rotor-mchlp2719/mac)

## PR Reviewer Guidelines
- It's best to pull the branch locally and test it, rather than just looking at the code online!
- Check that all code is adequately covered by tests - if it isn't feel free to suggest the addition of tests.
- Always make comments, even if it's minor, or if you don't understand why something was done.

### CODE REVIEW

**Review by ClayBenson94 — DISMISSED** [review]
https://github.com/CMSgov/mint-app/blob/7fe75289a2ff90c45f885662fa5a06e89de8ffc3/pkg/models/plan_payments.go#L25

I think this manually created go model might still need an update.

**Review by StevenWadeOddball — APPROVED** [review]
I tested by making a GQL mutation that sent every possible value for an enum array to this endpoint. It returned data successfully without any issue. Besides the item Clay discoverd, this should be good to go
```gql

mutation updatePlanPayments
{
    updatePlanPayments (id: "a18d2f8c-e5cb-4c5d-b21c-3554dbec9c51",changes:     
    {
        # Page 1
        fundingSource: [PATIENT_PROTECTION_AFFORDABLE_CARE_ACT,  TRUST_FUND,  OTHER]
        fundingSourceTrustFund: "hello"
        fundingSourceOther: "My cool note"
        fundingSourceNote: "ello"
        fundingSourceR: [PATIENT_PROTECTION_AFFORDABLE_CARE_ACT,  TRUST_FUND,  OTHER]
        fundingSourceRTrustFund: "trust me"
        fundingSourceROther: "Trust others"
        fundingSourceRNote: " Please trust us"
        payRecipients: [PROVIDERS, BENEFICIARIES, PARTICIPANTS, STATES, OTHER]
        payRecipientsOtherSpecification: "asdhjkasdlhjk"
        payRecipientsNote: "hasdjkaskhjd"
        payType: [CLAIMS_BASED_PAYMENTS, NON_CLAIMS_BASED_PAYMENTS, GRANTS]
        payTypeNote: "ashjkdasdhjkasd"

        # Page 2
        payClaims:   [ADJUSTMENTS_TO_FFS_PAYMENTS, CARE_MANAGEMENT_HOME_VISITS, SNF_CLAIMS_WITHOUT_3DAY_HOSPITAL_ADMISSIONS, TELEHEALTH_SERVICES_NOT_TRADITIONAL_MEDICARE, OTHER]
        # payClaimsOther:
        # shouldAnyProvidersExcludedFFSSystems:
        # shouldAnyProviderExcludedFFSSystemsNote:
        # changesMedicarePhysicianFeeSchedule:
        # changesMedicarePhysicianFeeScheduleNote:
        # affectsMedicareSecondaryPayerClaims:
        # affectsMedicareSecondaryPayerClaimsHow:
        # affectsMedicareSecondaryPayerClaimsNote:
        # payModelDifferentiation:

        # Page 3
        # creatingDependenciesBetweenServices:
        # creatingDependenciesBetweenServicesNote:
        # needsClaimsDataCollection:
        # needsClaimsDataCollectionNote:
        # providingThirdPartyFile:
        # isContractorAwareTestDataRequirements:

        # Page 4
        # beneficiaryCostSharingLevelAndHandling:
        # waiveBeneficiaryCostSharingForAnyServices:
        # waiveBeneficiaryCostSharingServiceSpecification:
        # waiverOnlyAppliesPartOfPayment:
        # waiveBeneficiaryCostSharingNote:

        # Page 5
        nonClaimsPayments:   [ADVANCED_PAYMENT,  BUNDLED_EPISODE_OF_CARE,  CAPITATION_POPULATION_BASED_FULL,  CAPITATION_POPULATION_BASED_PARTIAL,  CARE_COORDINATION_MANAGEMENT_FEE,  GLOBAL_BUDGET,  GRANTS,  INCENTIVE_PAYMENT,  MAPD_SHARED_SAVINGS,  SHARED_SAVINGS,  OTHER]
        # nonClaimsPaymentOther:
        # paymentCalculationOwner:
        # numberPaymentsPerPayCycle:
        # numberPaymentsPerPayCycleNote:
        # sharedSystemsInvolvedAdditionalClaimPayment:
        # sharedSystemsInvolvedAdditionalClaimPaymentNote:
        # planningToUseInnovationPaymentContractor:
        # planningToUseInnovationPaymentContractorNote:
        # fundingStructure:

        # Page 6
        expectedCalculationComplexityLevel: HIGH
        # expectedCalculationComplexityLevelNote:
        # canParticipantsSelectBetweenPaymentMechanisms:
        # canParticipantsSelectBetweenPaymentMechanismsHow:
        # canParticipantsSelectBetweenPaymentMechanismsNote:
        anticipatedPaymentFrequency:   [ANNUALLY, BIANNUALLY, QUARTERLY, MONTHLY, SEMIMONTHLY, WEEKLY, DAILY, OTHER]
        # anticipatedPaymentFrequencyOther:
        # anticipatedPaymentFrequencyNote:

        # Page 7
        # willRecoverPayments:
        # willRecoverPaymentsNote:
        # anticipateReconcilingPaymentsRetrospectively:
        # anticipateReconcilingPaymentsRetrospectivelyNote:
        # paymentStartDate:
        # paymentStartDateNote:

        # Meta
        # createdBy:
        # createdDts:
        # modifiedBy:
        # modifiedDts:
        # status:


    })
     {
  
    id
    modelPlanID

    # Page 1
    fundingSource
    fundingSourceTrustFund
    fundingSourceOther
    fundingSourceNote
    fundingSourceR
    fundingSourceRTrustFund
    fundingSourceROther
    fundingSourceRNote
    payRecipients
    payRecipientsOtherSpecification
    payRecipientsNote
    payType
    payTypeNote

    # Page 2
    payClaims
    payClaimsOther
    shouldAnyProvidersExcludedFFSSystems
    shouldAnyProviderExcludedFFSSystemsNote
    changesMedicarePhysicianFeeSchedule
    changesMedicarePhysicianFeeScheduleNote
    affectsMedicareSecondaryPayerClaims
    affectsMedicareSecondaryPayerClaimsHow
    affectsMedicareSecondaryPayerClaimsNote
    payModelDifferentiation

    # Page 3
    creatingDependenciesBetweenServices
    creatingDependenciesBetweenServicesNote
    needsClaimsDataCollection
    needsClaimsDataCollectionNote
    providingThirdPartyFile
    isContractorAwareTestDataRequirements

    # Page 4
    beneficiaryCostSharingLevelAndHandling
    waiveBeneficiaryCostSharingForAnyServices
    waiveBeneficiaryCostSharingServiceSpecification
    waiverOnlyAppliesPartOfPayment
    waiveBeneficiaryCostSharingNote

    # Page 5
    nonClaimsPayments
    nonClaimsPaymentOther
    paymentCalculationOwner
    numberPaymentsPerPayCycle
    numberPaymentsPerPayCycleNote
    sharedSystemsInvolvedAdditionalClaimPayment
    sharedSystemsInvolvedAdditionalClaimPaymentNote
    planningToUseInnovationPaymentContractor
    planningToUseInnovationPaymentContractorNote
    fundingStructure

    # Page 6
    expectedCalculationComplexityLevel
    expectedCalculationComplexityLevelNote
    canParticipantsSelectBetweenPaymentMechanisms
    canParticipantsSelectBetweenPaymentMechanismsHow
    canParticipantsSelectBetweenPaymentMechanismsNote
    anticipatedPaymentFrequency
    anticipatedPaymentFrequencyOther
    anticipatedPaymentFrequencyNote

    # Page 7
    willRecoverPayments
    willRecoverPaymentsNote
    anticipateReconcilingPaymentsRetrospectively
    anticipateReconcilingPaymentsRetrospectivelyNote
    paymentStartDate
    paymentStartDateNote

    # Meta
    createdBy
    createdDts
    modifiedBy
    modifiedDts
    status

    }
}
```

response
```gql
{
    "data": {
        "updatePlanPayments": {
            "id": "a18d2f8c-e5cb-4c5d-b21c-3554dbec9c51",
            "modelPlanID": "6e224030-09d5-46f7-ad04-4bb851b36eab",
            "fundingSource": [
                "PATIENT_PROTECTION_AFFORDABLE_CARE_ACT",
                "TRUST_FUND",
                "OTHER"
            ],
            "fundingSourceTrustFund": "hello",
            "fundingSourceOther": "My cool note",
            "fundingSourceNote": "ello",
            "fundingSourceR": [
                "PATIENT_PROTECTION_AFFORDABLE_CARE_ACT",
                "TRUST_FUND",
                "OTHER"
            ],
            "fundingSourceRTrustFund": "trust me",
            "fundingSourceROther": "Trust others",
            "fundingSourceRNote": " Please trust us",
            "payRecipients": [
                "PROVIDERS",
                "BENEFICIARIES",
                "PARTICIPANTS",
                "STATES",
                "OTHER"
            ],
            "payRecipientsOtherSpecification": "asdhjkasdlhjk",
            "payRecipientsNote": "hasdjkaskhjd",
            "payType": [
                "CLAIMS_BASED_PAYMENTS",
                "NON_CLAIMS_BASED_PAYMENTS",
                "GRANTS"
            ],
            "payTypeNote": "ashjkdasdhjkasd",
            "payClaims": [
                "ADJUSTMENTS_TO_FFS_PAYMENTS",
                "CARE_MANAGEMENT_HOME_VISITS",
                "SNF_CLAIMS_WITHOUT_3DAY_HOSPITAL_ADMISSIONS",
                "TELEHEALTH_SERVICES_NOT_TRADITIONAL_MEDICARE",
                "OTHER"
            ],
            "payClaimsOther": null,
            "shouldAnyProvidersExcludedFFSSystems": null,
            "shouldAnyProviderExcludedFFSSystemsNote": null,
            "changesMedicarePhysicianFeeSchedule": null,
            "changesMedicarePhysicianFeeScheduleNote": null,
            "affectsMedicareSecondaryPayerClaims": null,
            "affectsMedicareSecondaryPayerClaimsHow": null,
            "affectsMedicareSecondaryPayerClaimsNote": null,
            "payModelDifferentiation": null,
            "creatingDependenciesBetweenServices": null,
            "creatingDependenciesBetweenServicesNote": null,
            "needsClaimsDataCollection": null,
            "needsClaimsDataCollectionNote": null,
            "providingThirdPartyFile": null,
            "isContractorAwareTestDataRequirements": null,
            "beneficiaryCostSharingLevelAndHandling": null,
            "waiveBeneficiaryCostSharingForAnyServices": null,
            "waiveBeneficiaryCostSharingServiceSpecification": null,
            "waiverOnlyAppliesPartOfPayment": null,
            "waiveBeneficiaryCostSharingNote": null,
            "nonClaimsPayments": [
                "ADVANCED_PAYMENT",
                "BUNDLED_EPISODE_OF_CARE",
                "CAPITATION_POPULATION_BASED_FULL",
                "CAPITATION_POPULATION_BASED_PARTIAL",
                "CARE_COORDINATION_MANAGEMENT_FEE",
                "GLOBAL_BUDGET",
                "GRANTS",
                "INCENTIVE_PAYMENT",
                "MAPD_SHARED_SAVINGS",
                "SHARED_SAVINGS",
                "OTHER"
            ],
            "nonClaimsPaymentOther": null,
            "paymentCalculationOwner": null,
            "numberPaymentsPerPayCycle": null,
            "numberPaymentsPerPayCycleNote": null,
            "sharedSystemsInvolvedAdditionalClaimPayment": null,
            "sharedSystemsInvolvedAdditionalClaimPaymentNote": null,
            "planningToUseInnovationPaymentContractor": null,
            "planningToUseInnovationPaymentContractorNote": null,
            "fundingStructure": null,
            "expectedCalculationComplexityLevel": "HIGH",
            "expectedCalculationComplexityLevelNote": null,
            "canParticipantsSelectBetweenPaymentMechanisms": null,
            "canParticipantsSelectBetweenPaymentMechanismsHow": null,
            "canParticipantsSelectBetweenPaymentMechanismsNote": null,
            "anticipatedPaymentFrequency": [
                "ANNUALLY",
                "BIANNUALLY",
                "QUARTERLY",
                "MONTHLY",
                "SEMIMONTHLY",
                "WEEKLY",
                "DAILY",
                "OTHER"
            ],
            "anticipatedPaymentFrequencyOther": null,
            "anticipatedPaymentFrequencyNote": null,
            "willRecoverPayments": null,
            "willRecoverPaymentsNote": null,
            "anticipateReconcilingPaymentsRetrospectively": null,
            "anticipateReconcilingPaymentsRetrospectivelyNote": null,
            "paymentStartDate": null,
            "paymentStartDateNote": null,
            "createdBy": "FAKE",
            "createdDts": "2022-07-05T13:44:44.380055Z",
            "modifiedBy": "MINT",
            "modifiedDts": "2022-07-21T16:16:49.637878Z",
            "status": "READY"
        }
    }
}
```

**Comment by OddTomBrooks** [comment]
> https://github.com/CMSgov/mint-app/blob/7fe75289a2ff90c45f885662fa5a06e89de8ffc3/pkg/models/plan_payments.go#L25
> 
> I think this manually created go model might still need an update.

Good catch, fixed

**Review by StevenWadeOddball — APPROVED** [review]
looks good to me. I tested with the same mutation again

File: pkg/graph/schema.graphql [hunk 1]
```diff
@@ -1913,9 +1913,9 @@ enum PayType {
 
 enum ClaimsBasedPayType {
   ADJUSTMENTS_TO_FFS_PAYMENTS
-  PAYMENTS_CARE_MANAGEMENT_HOME_VISITS
-  PAYMENTS_SNF_CLAIMS_WITHOUT_3DAY_HOSPITAL_ADMISSIONS
-  PAYMENTS_TELEHEALTH_SERVICES_NOT_TRADITIONAL_MEDICARE
+  CARE_MANAGEMENT_HOME_VISITS
+  SNF_CLAIMS_WITHOUT_3DAY_HOSPITAL_ADMISSIONS
+  TELEHEALTH_SERVICES_NOT_TRADITIONAL_MEDICARE
   OTHER
```
Comment (garyjzhao) [inline pkg/graph/schema.graphql:1919]: There looks like some fields have gone missing, like `REDUCTIONS_TO_BENEFICIARY_COST_SHARING` and `  SERVICES_NOT_COVERED_THROUGH_TRADITIONAL_MEDICARE`

This was made evident to me through the merge conflicts

**Review by garyjzhao — CHANGES_REQUESTED** [review]
@OddTomBrooks Found a few missing field values

File: pkg/graph/schema.graphql (hunk 1 above)
Comment (ClayBenson94) [reply pkg/graph/schema.graphql:1919]: I see the same thing when looking at the merge conflicts. @OddTomBrooks can you look into this?

Comment (OddTomBrooks) [reply pkg/graph/schema.graphql]: Yep, it was just a bit off the head from main. Updated now.

**Review by StevenWadeOddball — APPROVED** [review]
lgtm, I tested with the mutation again after the changes

**Review by ClayBenson94 — APPROVED** [review]

**Pull request merged**